src/condor_utils/compat_classad.cpp

```diff
@@ -297,33 +300,32 @@ bool stringListSummarize_func( const char *name,
 		return false;
 	}
 
-	StringList sl( list_str.c_str(), delim_str.c_str() );
-	if ( sl.number() == 0 ) {
-		if ( empty_allowed ) {
-			result.SetRealValue( 0.0 );
-		} else {
-			result.SetUndefinedValue();
-		}
-		return true;
-	}
-
-	sl.rewind();
-	const char *entry;
-	while ( (entry = sl.next()) ) {
+	int cnt = 0;
+	for (auto& entry : StringTokenIterator(list_str, delim_str.c_str())) {
+		cnt++;
 		double temp;
-		int r = sscanf(entry, "%lf", &temp);
+		int r = sscanf(entry.c_str(), "%lf", &temp);
```

Comment:
Not all our compilers support this yet, but at some point, we should replace these sscanf's with the double overload of `std::from_chars`, which is faster, avoids all format-string security and varags wonkiness.